core/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java

```diff
@@ -316,6 +316,18 @@ public void writeStringArrayNullable(@Nullable String[] array) throws IOExceptio
         }
     }
 
+    /**
+     * Writes a string array, for nullable string, writes it as 0 (empty string).
+     */
```

Comment:
Comment seems wrong, as far as I can see the difference to `writeStringArrayNullable` is that it writes boolean instead of int in case of null argument.